GameBoard/Gomoku/GomokuBoard.cs

```diff
@@ -0,0 +1,113 @@
+using System.Data;
+using System.Security.Cryptography;
+using System.Text.Json;
+using GameBoard;
+
+namespace Gomoku;
+
+public class GomokuBoard : GameBoard.GameBoard
+{
+    protected override int PlayerCount => 2;
+    protected override string GameRecordFileName => "gomoku-record.json";
+    public override string GameName => "Gomoku";
+    private const char FirstPlayerSymbol = 'o';
+    private const char SecondPlayerSymbol = 'x';
+
+    static GomokuBoard()
+    {
+        GameBoardFactory.RegisterGame(() => new GomokuBoard());
+    }
+    
+    protected override void SetupGameBoard()
+    {
+        Size = 15;
+
+        base.SetupGameBoard();
+    }
+
+    public override bool CheckWin(int row, int col, object value = null)
+    {
+        const int minEdge = 0;
+        const int winningCount = 5;
+        if (value == null)
+        {
+            value = Board[row, col];
+            if (value == null) { return false; }
+        }
+        //Horizontal line
+        int count = 1;
+        for (int distance = 1; distance < winningCount; distance++)
+        {
+            if (col + distance < Size && Board[row, col + distance] != null && Board[row, col + distance].Equals(value)) { count++; }
+            else { break; }
+        }
+        for (int distance =1; distance <winningCount; distance++)
```

Comment:
These two for loops look similar to each other; and if we combine them so that we can calculate the max length of same symbols in a row.

Is it simpler for you?

```suggestion
        for (int distance =1; distance <winningCount; distance++)
        
        int maxLengthInARow = 0;
        for(int i = col - distance; i <= col + distance; i ++) {
           if(i < minEdge || i > Size) continue;
           
           if(Board[row, col + i].Equals(value) { count++; }
           else
           {
               maxLengthInARow = Math.max(maxLengthInARow, count);
               count = 0;
           }
        }
        if (maxLengthInARow >= winningCount) { return true; }
        ```